solr/core/src/java/org/apache/solr/search/CaffeineCache.java

```diff
@@ -211,6 +211,7 @@ private V computeAsync(K key, IOFunction<? super K, ? extends V> mappingFunction
       try {
         // Another thread is already working on this computation, wait for them to finish
         V value = result.join();
+        if (value == null) return null;
```

Comment:
Separate issue?

Reply:
This is arguably a proper change in principle, but it's vestigial here in the sense that the need that caused this issue to become apparent is no longer relevant. I should remove this, and probably at some point pursue it upstream. Good catch.